wordpress-mobile/WordPress-Android
# Login rework: prologue screen

As we start to implement the rework of the login flows (aka, the mobile-login project), let's start with implementing the very first screen, the prologue screen, where the user get's asked whether they want to log in or sign up.

This PR does not focus on the visual design side as that will be worked on in a subsequent PR.

At the moment, the screen doesn't actually implement any of the functionality presented to the user. Instead, it shows a Toast message that login/signup is not implemented yet. The actual implementation will be done in a subsequent PR.

There are a few unit tests included that test the navigation logic. Almost all tests are basically testing some trivial things, like: login is not implemented so the state machine needs to still be at the initial state.

All in all, this PR tries to lay the foundation upon which to add the real implementation of the prologue and the rest of the screens. Attention has been put on these features:

* Add a feature flag for the new login flows (`LOGIN_WIZARD_STYLE_ACTIVE`) so we can merge new functionality directly in `develop` and easily follow any developments in `develop`. The flag is set to `true` for the dev (`wasabi`) build only. Production/Alpha/Beta builds have it set to `false` for now.

To test:
* Run the `wasabi` build. Have the app data cleaned up or just uninstall/install
* Notice the new login screen, asking you to either choose to log in or sign up. Tap any of the buttons
* A `... is not implemented yet` Toast comes up

As part of testing this PR, please have a look at the way the fragment (`LogInOrSignUpFragment`) communicates with the parent activity (`LoginActivity`) in a format modular enough so the navigation can be tested with Robolectric (unit) tests.

To run the tests through AndroidStudio, open the `LoginLogicTest.java` file, right-click on the class name and click on "Run LoginLogicTest". If the run fails, make sure the run configuration has `-ea` in the VM Options and that the "Working directory" points to `$MODULE_DIR`.
